addons/website/static/src/snippets/s_dynamic_snippet/000.scss

```diff
@@ -46,4 +46,13 @@
             }
         }
     }
+
+    .dynamic_snippet_template .col-12:not(:has(.o_template_product_product_banner)) {
+    // When a single element has been fetched, it takes up the entire width and becomes
+    // too tall according to the page height.
+    // This forces the same size to be used as when two elements are fetched.
+        @include media-breakpoint-up(lg) {
+            max-width: 50%;
+        }
+    }
```

Comment:
If we want to use `col-lg-6` in this case, I think we'd have to modify the calculation made here :
https://github.com/odoo/odoo/blob/18.0/addons/website/static/src/snippets/s_dynamic_snippet_carousel/000.xml#L18

Moreover, this change would have to be made in all templates except the "banner" one, so I'm not sure it's easily doable by modifying this calculation in the short time we have 🤔 .

That's why I preferred to use SCSS, as it's the easiest way to achieve this rendering.

The best solution would be to provide templates that work with only one fetched element, but I guess this is something we could do later 🙂

Reply:
> If we want to use col-lg-6 in this case, I think we'd have to modify the calculation made here :

Seems like what we should do indeed :+1: 

> Moreover, this change would have to be made in all templates except the "banner" one

I am not really familiar with this dynamic snippet system but I don't think so, is not that just about setting a "minRowSize" (or anything like that) in the code you link and changing its value *only for* the banner one?

> not sure it's easily doable by modifying this calculation in the short time we have 

There is no rush, we can fix it the right way, it's not a critical bug. Also if there is a rush, it would way quickier to just force banner to act like there other ones, changing only one line in the code (the one you linked) and improve later so that banner can be full-width :pray: 

> That's why I preferred to use SCSS, as it's the easiest way to achieve this rendering.

Adding SCSS is never the easiest for me for as it forces me to think about upgrades if I ever was to remove that CSS in the future :wink: And when there is a `:has` in there, it's really a proof that something wrong is going on.

Reply:
> There is no rush, we can fix it the right way, it's not a critical bug. 

Yes you're right 👍 
We'll probably review this snippet, providing more compatible templates with a single item, later after the OXP.
I'll close this PR.

Thanks again for the review 🙏